### PR TITLE
Feature/add message count

### PR DESCRIPTION
**Pull request by mikeSmith08**
Update message delivery count and status when processing is not successful, or a status of not COMPLETED is returned from a message type processor.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Message.java [hunk 1]
```diff
@@ -30,4 +31,9 @@
     @Column(name = "message_payload")
     private String messagePayload;
 
+    @Column(name = "delivery_count")
+    private int deliveryCount;
+
+    @Column(name = "status")
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Message.java]: Need an enumerated annotation.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java [hunk 2]
```diff
@@ -75,18 +78,33 @@ private MessageStatus processMessage(Message message, MessageTypeProcessor messa
...
-            // TODO: HTBHF-1285: handle other message statuses
+
             return status;
         } catch (RuntimeException e) {
             // TODO: HTBHF-1285: expose ErrorHandler.constructExceptionDetail
             //                  from java common and include it in error handling
             log.error("Unable to process message with id {}", message.getId(), e);
-            // TODO: HTBHF-1285: increment delivery count and message timestamp
-            return MessageStatus.ERROR;
+            MessageStatus errorStatus = ERROR;
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: We can replace these two lines with updateMessagesToErrorAndIncrementCount

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Message.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Message.java]: Done.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: That method can validly take different statuses so I'll leave it like that. I've also added ERROR as a variable as this is what's returned from this method.

**Pull request merged**